internal/debug/debug.go

```diff
@@ -0,0 +1,165 @@
+package debug
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os/exec"
+	"strings"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+type DebugPod struct {
+	Name      string
+	Namespace string
+	NodeName  string
+}
+
+const (
+	interval       = time.Millisecond * 500
+	duration       = time.Second * 5
+	debugNamespace = "openshift-debug"
+)
+
+// New creates debug pod on the given node, puts it in infinite sleep,
+// and returns the DebugPod object. Use the Clean() method to delete it.
+func New(node string) (*DebugPod, error) {
+	errCh := make(chan error)
+	go func() {
+		errCh <- start(node)
+	}()
+	for {
+		select {
+		case err := <-errCh:
+			return nil, err
+		default:
+			return waitForPod()
+		}
+	}
+}
+
+func (dp *DebugPod) Exec(cmd string) ([]byte, error) {
+	cmdOnDebugPod := append([]string{"exec", "-n", dp.Namespace, dp.Name, "--", "chroot", "/host"}, strings.Split(cmd, " ")...)
+	out, err := exec.Command("oc", cmdOnDebugPod...).CombinedOutput()
+	if err != nil {
+		return nil, fmt.Errorf("failed to exec command \"%s\" on node %s: %v\n%s", cmd, dp.NodeName, err, string(out))
+	}
+
+	return out, nil
+}
+
+func (dp *DebugPod) ExecWithRetry(cmd string) ([]byte, error) {
```

Comment:
nit: I think you should accept the interval/duration here and not hardcode it

Reply:
done